### ISSUE TITLE
@EnsuresQualifier / @EnsuresNonull should apply only to Class fields

### ISSUE THREAD

**Issue by the1derer** [typetools/checker-framework#2479]
In the following example:
```java
import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
import org.checkerframework.checker.nullness.qual.EnsuresNonNull;
import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.framework.qual.EnsuresQualifier;

class Test {
    public @MonotonicNonNull String field1;

    // @EnsuresQualifier(expression = "field1", qualifier = NonNull.class)
    @EnsuresNonNull("field1")
    public void set(String field1) {
        this.field1 = field1;
    }

}
```
Using Nullness Checker I am receiving the warning:
```error
Test.java:12: warning: [contracts.postcondition.expression.parameter.name] The postcondition EnsuresQualifier on the declaration of method 'set' refers to formal parameter 'field1' by name. Please use "#1" instead of "field1".
    public void set(String field1) {
                ^
1 warning
```
If I replace ``@EnsuresNonNull("field1")`` by ``@EnsuresNonNull("this.field1")`` the check is passed.
But the warning should not be given in first place as ``@EnsuresNonNull`` should only be applied to fields.
